.editorconfig

```diff
@@ -8,255 +8,596 @@ end_of_line = LF
 [*.cs]
 indent_style = tab
 indent_size = 4
 
-dotnet_separate_import_directive_groups = false
-dotnet_sort_system_directives_first = true
+#### Code Style Rules
+#### https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/
 
-csharp_style_var_elsewhere = true:suggestion
-csharp_style_var_for_built_in_types = true:suggestion
-csharp_style_var_when_type_is_apparent = true:suggestion
+# Severity Levels: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-options#severity-level
+# Below we enable specific rules by setting severity to warning.
+# Rules are disabled by setting severity to silent (to still allow use in IDE) or none (to prevent all use).
+# Rules are listed below with any options available.
+# Options are commented out if they match the defaults.
 
-csharp_prefer_braces = when_multiline:suggestion
-csharp_using_directive_placement = outside_namespace:suggestion
-csharp_new_line_before_open_brace = all
-csharp_space_around_binary_operators = before_and_after
+### Language Rules
+### https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/language-rules
 
-## Naming styles:
+## this and Me preferences
 
-dotnet_naming_style.camel_case.capitalization = camel_case
+# IDE0003/IDE0009 Remove 'this' or 'Me' qualification/Add 'this' or 'Me' qualification
+#dotnet_style_qualification_for_field = false
+#dotnet_style_qualification_for_property = false
+#dotnet_style_qualification_for_method = false
+#dotnet_style_qualification_for_event = false
+dotnet_diagnostic.IDE0003.severity = warning
+dotnet_diagnostic.IDE0009.severity = warning
 
-dotnet_naming_style.pascal_case.capitalization = pascal_case
+## Use languages keywords for types
 
-dotnet_naming_style.i_prefix_pascal_case.capitalization = pascal_case
-dotnet_naming_style.i_prefix_pascal_case.required_prefix = I
+# IDE0049 Use language keywords instead of framework type names for type references
+#dotnet_style_predefined_type_for_locals_parameters_members = true
+#dotnet_style_predefined_type_for_member_access = true
+dotnet_diagnostic.IDE0049.severity = warning
 
-## Symbol specifications:
+## Modifier preferences
 
-dotnet_naming_symbols.const_locals.applicable_kinds = local
-dotnet_naming_symbols.const_locals.applicable_accessibilities = *
-dotnet_naming_symbols.const_locals.required_modifiers = const
+# IDE0036 Order modifiers
+#csharp_preferred_modifier_order = public, private, protected, internal, file, static, extern, new, virtual, abstract, sealed, override, readonly, unsafe, required, volatile, async
+dotnet_diagnostic.IDE0036.severity = warning
 
-dotnet_naming_symbols.const_fields.applicable_kinds = field
-dotnet_naming_symbols.const_fields.applicable_accessibilities = *
-dotnet_naming_symbols.const_fields.required_modifiers = const
+# IDE0040 Add accessibility modifiers
+dotnet_style_require_accessibility_modifiers = omit_if_default #for_non_interface_members
```

Comment:
`#for_non_interface_members` is that to be enabled in the future?

Reply:
No, it's a comment to indicate the default value of the setting. Since OpenRA prefers to have implicit accessibility eveyrwhere, we change to `omit_if_default` instead.

Reply:
Is there really a reason to list the default value? I don't see the benefit of that and it is just confusing because you don't know if that is a future TODO.

Reply:
If something is a TODO then it will/should say "TODO: ...".
But I do agree specifying the default values that we override is redundant.

Reply:
Removed the comments when defaults overridden.

Reply:
if we aren't worried that this will become a maintenance overhead, I'm all for displaying default values. It saves a bunch of hassle when working with rules

Reply:
What is your concrete usecase that would be a hassle? If we don't use the default then there is a reason for it. And if not this should be a TODO.

Reply:
it's great for when we want to change rules again, whenever that will be